pkg/config/validation/validation.go

```diff
@@ -2492,9 +2492,10 @@ func validateTLSMatch(match *networking.TLSMatchAttributes, context *networking.
 func validateSniHost(sniHost string, context *networking.VirtualService) error {
 	if err := ValidateWildcardDomain(sniHost); err != nil {
 		ipAddr := net.ParseIP(sniHost) // Could also be an IP
-		if ipAddr == nil {
-			return err
+		if ipAddr != nil {
+			return fmt.Errorf("IP address %q is not a valid SNI host value", ipAddr)
```

Comment:
I think we should make it a warning. I agree it matches the spec but its a breaking change if we do this; users could already have these configs and they will start breaking.

Reply:
@howardjohn what do you think about the wording for the warning?
```
not all clients support IP address as an SNI value, use with caution
```

Reply:
Does Envoy even support it? Its against SNI spec, so I think we should be a bit more strongly worded

Reply:
so currently in Istio, if you specify `sniHosts` with an IP address you do get a listener with the proper SNI value set:
```
0.0.0.0   443  SNI: 1.2.3.4     Cluster: outbound|443|cnn|istio-egressgateway.istio-system.svc.cluster.local
```
So Envoy accepts an IP address in the generated XDS.  Using `curl` as the client still results in a request going through PassthroughCluster
```
[2022-02-28T21:34:00.905Z] "- - -" 0 UF,URX - - "-" 0 0 10015 - "-" "-" "-" "-" "1.2.3.4:443" PassthroughCluster - 1.2.3.4:443 172.30.141.10:59494 - -
```
I tried to configure a Go client example with IP based SNI but it too still goes through `PassthroughCluster`, so I'm not sure what client to test if Envoy functionally supports them

Reply:
latest message:
```
using an IP address (%q) goes against SNI spec and most clients do no support this
```